### PR TITLE
[refactor]: Multihash display format

### PR DESCRIPTION
**Pull request by mversic**
### Description of the Change

keys are encoded uppercase, with a caveat that for multihash the fn_code is lowercase
improved private key validation

### Issue

following [this](https://github.com/hyperledger/iroha/pull/3077#discussion_r1141700909) discussion

### CODE REVIEW

**Review by Erigara — COMMENTED** [review]
Will this format of key representation be familiar and convenient for the end user?
I checked substrate and ethereum blockchain explorers and there key display looks more like our initial display format.

File: schema/gen/src/lib.rs [hunk 1]
```diff
@@ -606,4 +606,13 @@ mod tests {
             "Missing schemas: \n{missing_schemas:#?}"
         );
     }
+
+    #[test]
+    // NOTE: This test guards from incorrect implementation where
+    // `SortedVec<T>` and `Vec<T>` start stepping over each other
+    fn no_schema_type_overlap() {
```
[the remark this answers is not in the document]
Comment (mversic) [reply schema/gen/src/lib.rs:613]: yes, this test will panic if implementation is incorrect. This test covers the reason why I went from internal field in `Vec::sorted` to two schema types `SortedVec`/`Vec`

**Comment by mversic** [comment]
> Will this format of key representation be familiar and convenient for the end user? I checked substrate and ethereum blockchain explorers and there key display looks more like our initial display format.

That is the question. The point was to make keys easily visually discernible. It's just whitespace injection:
* to separate private vs public key in the private key
* to separate different parts of the multihash format

If you think this is not more user friendly, I will close the PR

**Comment by mversic** [comment]
> I have a bit general question not directly tied up to this PR, but as you wrote in PR description we have public and private parts in private key. Do we really need public part? Because in this case our `KeyPair` type seems redundant (public, public, private).

it depends on the algorithm used. `KeyPair` is not always redundant. Only when `Ed25519` algorithm is used does private key contain public key

**Comment by Erigara** [comment]
> I just wanted others to asses if this is a good idea

I agree let's gather other opinion. @pesterev @ilchu @appetrosyan what do you think about key display format? 

> There is less dilema if the key is enclosed in quotes, which it (always?) is

Yeah, i see now, i was thinking about `kagami` initially but it seems to use quotes anyway (at least by default and in json formats).

**Comment by mversic** [comment]
> Multihash is a relatively widely used format:
> 
>     * [multiformats/multihash](https://github.com/multiformats/multihash)
> 
> 
> I am not sure that deviation from the standard display format (which is currently used by in Iroha) is a good idea.

this is the page that I was looking at. Look [here](https://raw.githubusercontent.com/multiformats/multihash/master/img/multihash.006.jpg), different parts are colored differently. I can't use colors so I separated them with spaces

Still, I'm not saying this is better I wanted to hear different opinions

**Review by Erigara — DISMISSED** [review]
LGTM

**Review by Erigara — APPROVED** [review]

**Review by 0x009922 — APPROVED** [review]

**Pull request merged**